addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusBaseThingHandler.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.fronius.handler;
+
+import java.math.BigDecimal;
+import java.util.Calendar;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.fronius.FroniusBridgeConfiguration;
+import org.openhab.binding.fronius.api.ValueUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Basic Handler class for all Fronius services.
+ *
+ * @author Gerrit Beine - Initial contribution
+ * @author Thomas Rokohl - Refactoring to merge the concepts
+ */
+public abstract class FroniusBaseThingHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FroniusBaseThingHandler.class);
+    private final String serviceDescription;
+
+    public FroniusBaseThingHandler(Thing thing) {
+        super(thing);
+        serviceDescription = getDescription();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command {} is not supported for channel: {}", command, channelUID.getId());
+    }
+
+    @Override
+    public void initialize() {
+        if (getBridge() == null) {
+            logger.error("Initializing {} Service is only supported within a bridge", serviceDescription);
+            updateStatus(ThingStatus.OFFLINE);
```

Comment:
Please also check if you have a `FroniusBridgeHandler` and if not set the thing to OFFLINE with reason `BRIDGE_OFFLINE`. For an example see https://github.com/eclipse/smarthome/blob/master/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java#L119